project/src/components/app/app.jsx

```diff
@@ -1,13 +1,45 @@
-import React from 'react';
+import React, { Fragment } from 'react';
 import PropTypes from 'prop-types';
-import WelcomeScreen from '../wellcome-screen/wellcome-screen';
+import { Switch, Route, BrowserRouter, Link } from 'react-router-dom';
+import { AppRoute } from '../../const';
+import Main from '../main/main';
+import SingIn from '../sign-in/sign-in';
+import Favorites from '..//favorites/favorites';
+import Room from '../room/room';
 
 
 function App(props) {
-  const {cardsCount} = props;
+  const { cardsCount } = props;
 
   return (
-    <WelcomeScreen cardsCount={cardsCount}/>
+    <BrowserRouter>
+      <Switch>
+        <Route exact path={AppRoute.MAIN}>
+          <Main cardsCount={cardsCount} />
+        </Route>
+        <Route exact path={AppRoute.SIGN_IN}>
+          <SingIn />
+        </Route>
+        <Route exact path={AppRoute.FAVORITES}>
+          <Favorites />
+        </Route>
+        <Route exact path={AppRoute.ROOM}>
+          <Room />
+        </Route>
+        <Route
+          render = {() => (
+            <Fragment>
+              <h1>
+                404.
+                <br />
+                <small>Page not found</small>
+              </h1>
+              <Link to="/">Go to main page</Link>
+            </Fragment>
```

Comment:
```suggestion
            <>
              <h1>
                404.
                <br />
                <small>Page not found</small>
              </h1>
              <Link to="/">Go to main page</Link>
            </>
```
Отдавай предпочтение такому синтаксису фрагментов

А еще было-бы лучше сделать отдельный компонент для этого